### ISSUE TITLE
Smartedit slot contextual menu missing due to missing CSS on content slots

### ISSUE THREAD

**Issue by rirypheng** [SAP/cloud-commerce-spartacus-storefront#845]
CSS changes needed to address the issue described in https://jira.hybris.com/browse/CMSX-8203.
The width and height of some content slots are shown as 0px. Therefore, smartedit is not able to render the contextual menu for those slots correctly.

Note: if you point at bottom banner they start to show up, because the width/height are there. then it allows others to work.

**Comment by tobi-or-not-tobi** [comment]
It's recommended to wait until we've done the changes in #950 as the page template and slot css will be changed.

**Comment by tobi-or-not-tobi** [comment]
@rirypheng the storefront is not showing empty slots, as it's meant to not use any space. The UI will be destroyed if we would render empty slots. 

I can only imagine 2 ways to allow business users to add content to empty slots:
- the Spartacus css layer identifies we're in SmartEdit and in _edit_ mode, and will therefor apply a min-width and min-height to the slot. 
- SmartEdit allows for assigning content to empty (not visible) slots, by providing a list of page(template) slots, outside the visual presentation.

**Comment by tobi-or-not-tobi** [comment]
@rirypheng says:

> I was not able to reproduce the issue for the ElectronicsHompageSplashBannerComponent in Section1Slot-Homepage as reported in https://jira.hybris.com/browse/CMSX-8203. I tried it locally, by running the latest spa code, and on your demo server https://dev-com-2.accdemo.b2c.ydev.hybris.com:9002/smartedit.

This is most likely caused by changes coming from #907. The configurable page layouts have turned our slot elements into _block_ elements (before they were _inline_). A block element will take 100% of the size by default. 
The footer (and header) is not yet rendered by page layout, but that will change soon. 

@rirypheng can you confirm that the block element is what we need here? (you might want to change it in dev tools to confirm). if so, please close this ticket.

**Comment by rirypheng** [comment]
@tobi-or-not-tobi the footer navigation slot was detected properly after adding the block element as suggested. Closing this ticket as it will be fixed in upcoming changes.

**Comment by Xymmer** [comment]
will be fixed with a new smartedit release

**Comment by tobi-or-not-tobi** [comment]
@Xymmer it will be fixed once we have all our page templates and header/footer being rendered using our page layouts. That change will ensure that all slots are no longer rendered as an inline element. 

We should start some best practice documentation to help customers understand about some details around spartacus / smartedit integration.

**Comment by rirypheng** [comment]
The issue is reproducible again on the homepage for the bottomHeaderSlot using the latest codes. Please see attached images. 
![misplaced slot](https://user-images.githubusercontent.com/17711826/54129650-1cb97a00-43e5-11e9-8486-ac10b5d0afcf.png)
![console](https://user-images.githubusercontent.com/17711826/54129649-1cb97a00-43e5-11e9-9f58-6221759ea247.png)

**Comment by developpeurweb** [comment]
Hi @tobi-or-not-tobi this ticket was just brought to my attention, since it's related to the navigation menu, i was just wondering if this bug could be related to the **refactoring effort** concerning this other ticket GH-607 and possibly these files below as well?
`projects/storefrontlib/src/lib/cms-lib/navigation/navigation-ui.component.scss`
`projects/storefrontlib/src/lib/cms-lib/navigation/navigation-ui.component.html`

**Comment by stepan-aleksandrov** [comment]
On the latest develop the following css breaks smartedit:
```
.ContentPage1Template cx-page-slot.Section2A, .MultiStepCheckoutSummaryPageTemplate cx-page-slot.Section2A, .ContentPage1Template cx-page-slot.Section2B, .MultiStepCheckoutSummaryPageTemplate cx-page-slot.Section2B {
    display: block;
}
```
Removing  **display: block;** fixes it.

**Comment by tobi-or-not-tobi** [comment]
@rirypheng The issue related to the BottomHeaderSlot will be fixed as part of #1985. 

I've noticed based on this ticket and #1985 that besides the contract, the CSS layer needs to make slots in block mode, or at least make sure that the slot element has a certain width. This is a requirement for storefronts, that might need more attention in the documentation (or I've missed it). 

The problem with modern frontend code, is that (web)components are typically created with custom selectors, rather than the traditional `DIV` element. A DIV element is rendered in block mode by default in browsers, while custom selectors are rendered inline (similar to a `SPAN` element). The CSS layer needs to make the custom elements in block mode explicitly. 

It would be nice if SmartEdit would be able to generate the overlay element without this constraint. I can imagine, that instead of using the slot size as an input to calculate the overlay, the dimensions of the embedded components would be taken into account. Just a thought. Could also be used in case the slot is not having any dimensions to calculate with.

**Comment by Xymmer** [comment]
so.......... looks like it's fixed (menu is showing) but it's not behaving properly, we'll create a new ticket to investigate.
